src/libs/Log.js

```diff
@@ -4,22 +4,28 @@ import CONFIG from '../CONFIG';
 import getPlatform from './getPlatform';
 import {version} from '../../package.json';
 import NetworkConnection from './NetworkConnection';
+import HttpUtils from './HttpUtils';
+
+let timeout = null;
+let info;
 
 /**
  * Network interface for logger.
  *
  * @param {Object} params
  * @param {Object} params.parameters
  * @param {String} params.message
+ * @return {Promise}
  */
 function serverLoggingCallback(params) {
-    const requestParams = {
-        message: params.message,
-        parameters: JSON.stringify(params.parameters || {}),
-        expensifyCashAppVersion: `expensifyCash[${getPlatform()}]${version}`,
-    };
-
-    API.Log(requestParams);
+    const requestParams = params;
+    requestParams.expensifyCashAppVersion = `expensifyCash[${getPlatform()}]${version}`;
+    if (requestParams.parameters) {
+        requestParams.parameters = JSON.stringify(params.parameters);
+    }
+    clearTimeout(timeout);
+    timeout = setTimeout(() => info('Flushing logs older than 10 minutes', true, {}, true), 10 * 60 * 1000);
```

Comment:
> What is the significance of 10 minutes? 

I picked 10 randomly, we can change it if we want.

> Does this basically mean we will have to wait a whole ten minutes to see any logs under 50? 

Kind of yes, at most 10 minutes, because you are missing in your `And so we flush logs when` that when you log alert or warning we flush logs immediately (since we assume we'll be more interested in errors and warnings than infos)

> Seems like a potentially long time to wait if you are trying to figure something out in realtime, but hard to say.

In real time while a user is doing things or for your own account? In any case, due to all the conditions mentioned above I don't think this will matter much.

> Also having trouble wrapping my head around how the network queue will handle the new logging situation when we are offline... Should we test those scenarios?

Good question, yes, I should definitely test that. I thought it would stay in the queue and send when going online, but will do some tests

Reply:
>I picked 10 randomly, we can change it if we want.

I think it's fine now that I understand what you are saying about warnings flushing immediately.

>In real time while a user is doing things or for your own account?

Could be either but I was thinking of the case where Jason or someone says, "it just did the thing where it breaks just now go check logs to see" and maybe we are having to wait 10 minutes because no error was thrown. But I'm not sure this is a real problem - since an error _probably_ will be thrown.

Reply:
Yeah, I don't think that will be a problem, as soon as Jason goes to slack to report the error, the logs will be sent because he backgrounded the app 😄